### PR TITLE
Add libhyper (to eventually add into curl).

### PR DESCRIPTION
**Pull request by dlorenc**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates

### CODE REVIEW

File: libhyper.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  - runs: |
+      rustup toolchain add nightly
+      export PATH=$HOME/.rustup/toolchains/nightly-${{host.triplet.rust}}/bin:$PATH
+      RUSTFLAGS="--cfg hyper_unstable_ffi" \
+        cargo rustc --features client,http1,http2,ffi -Z unstable-options \
+        --crate-type cdylib --release
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/lib
+      mv target/release/libhyper.so ${{targets.destdir}}/usr/lib/
```
Comment (kaniini) [inline libhyper.yaml:35]: Shouldn't this have an SONAME or something?

**Review by rawlingsj — DISMISSED** [review]

File: libhyper.yaml (hunk 1 above)
Comment (dlorenc) [reply libhyper.yaml:35]: You mean like a libhyper.so.0 or something?

Comment (kaniini) [reply libhyper.yaml:30]: Yeah, exactly.  And linked with the `-soname` LDFLAG to ensure the dependency name is set up correctly.

Comment (dlorenc) [reply libhyper.yaml]: How's this?

File: libhyper.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  - runs: |
+      RUSTFLAGS="--cfg hyper_unstable_ffi -Clink-arg=-Wl,-soname=libhyper.so.0" cargo rustc --features client,http1,http2,ffi --crate-type cdylib --release
+
+      # Need to manually set soname since cargo doesn't support that
+      patchelf --set-soname libhyper.so.0 target/release/libhyper.so
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/lib
+      mv target/release/libhyper.so ${{targets.destdir}}/usr/lib/libhyper.so.0
```
Comment (kaniini) [inline libhyper.yaml:34]:
```suggestion
      mv target/release/libhyper.so ${{targets.destdir}}/usr/lib/libhyper.so.0.14.27
```

File: libhyper.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      ln -s libhyper.so.0 ${{targets.destdir}}/usr/lib/libhyper.so
+      ln -s libhyper.so.0 ${{targets.destdir}}/usr/lib/libhyper.so.0.14.27
```
Comment (kaniini) [inline libhyper.yaml:36]:
```suggestion
      ln -s libhyper.so.0.14.27 ${{targets.destdir}}/usr/lib/libhyper.so.0
```

**Review by kaniini — CHANGES_REQUESTED** [review]

**Review by kaniini — APPROVED** [review]

**Pull request merged**